src/OrchardCore.Modules/OrchardCore.Users/Deployment/CustomUserSettingsDeploymentSource.cs

```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Deployment;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using YesSql;
+
+namespace OrchardCore.Users.Deployment;
+
+public class CustomUserSettingsDeploymentSource : IDeploymentSource
+{
+    private readonly CustomUserSettingsService _customUserSettingsService;
+    private readonly ISession _session;
+
+    public CustomUserSettingsDeploymentSource(CustomUserSettingsService customUserSettingsService, ISession session)
+    {
+        _customUserSettingsService = customUserSettingsService;
+        _session = session;
+    }
+
+    public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+    {
+        var customUserSettingsStep = step as CustomUserSettingsDeploymentStep;
+        if (customUserSettingsStep == null)
+        {
+            return;
+        }
+
+        var settingsTypes = customUserSettingsStep.IncludeAll
+            ? _customUserSettingsService.GetAllSettingsTypes().ToArray()
+            : _customUserSettingsService.GetSettingsTypes(customUserSettingsStep.SettingsTypeNames).ToArray();
+
+        // Todo: check permissions for each settings type
+        var bigArray = new JArray();
+        var allUsers = await _session.Query<User>().ListAsync();
+
+        foreach (var user in allUsers)
+        {
+            var myArray = new JArray();
+            foreach (var settingsType in settingsTypes)
+            {
+                var userSetting = await _customUserSettingsService.GetSettingsAsync(user, settingsType);
+                myArray.Add(JObject.FromObject(userSetting));
+            }
+
+            bigArray.Add(new JObject(
+                new JProperty("userId", user.UserId),
+                new JProperty("user-custom-user-settings", myArray)));
```

Comment:
maybe defined the `user-custom-user-settings` and `custom-user-settings` as local private constants.

Reply:
I don't know, this is the kind of convetion use3d in other deployments

Reply:
I think that's fine as it is.

More globally I think we should not block a PR for things that are done in the same way in all other related places, I think that's a good convention to trust existing code that may respect things that we are not aware, or suggest a separate PR to update them all if there is an issue or just to suggest a better way.